Amazon.IonDotnet.Tests/Internals/BinaryWriterTest.cs

```diff
@@ -450,5 +452,36 @@ void writeAndAdd<T>(string fieldName, T value, Action<T> writeAction)
 
             return kvps;
         }
+
+        [TestMethod]
+        public void WriteValuesWithAnnotation()
+        {
+            var ionValues = new List<IIonValue>();
+            var valueFactory = new ValueFactory();
+            ionValues.Add(valueFactory.NewBool(true));
+            ionValues.Add(valueFactory.NewString("string"));
+            ionValues.Add(valueFactory.NewFloat(22.22f));
+            ionValues.Add(valueFactory.NewDecimal(3.4m));
+            ionValues.Add(valueFactory.NewInt(3));
+            ionValues.Add(valueFactory.NewNull());
+            ionValues.Add(valueFactory.NewSymbol("symbol"));
+            ionValues.Add(valueFactory.NewTimestamp(new Timestamp(DateTime.Now)));
+            var arrayOfbytes = Enumerable.Repeat<byte>(1, 10).ToArray();
+            ReadOnlySpan<byte> bytes = new ReadOnlySpan<byte>(arrayOfbytes);
+            ionValues.Add(valueFactory.NewClob(bytes));
+            ionValues.Add(valueFactory.NewBlob(bytes));
+
+            foreach (var ionValue in ionValues)
+            {
+                ionValue.AddTypeAnnotation("annotation");
+            }
+
+            using var writer = IonBinaryWriterBuilder.Build(new MemoryStream());
+            foreach (var ionValue in ionValues)
+            {
+                ionValue.WriteTo(writer);
+            }
+            writer.Finish();
+        }
```

Comment:
The test is missing an assertion that the writer produced the expected result.

Reply:
Oh my bad. Updated the test.